daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala

```diff
@@ -889,7 +900,218 @@ class EvaluationOrderTest extends AnyFreeSpec with Matchers with Inside {
       "key exceeds max nesting" in {
         val (res, msgs) = evalUpdateApp(
           pkgs,
-          e"""\(sig : Party) (fetcher: Party) -> Test:fetch_by_key fetcher (Test:someParty sig) Test:noCid 100""",
+          e"""\(sig : Party) (fetchingParty: Party) -> Test:fetch_by_key fetchingParty (Test:someParty sig) Test:noCid 100""",
+          Array(SParty(alice), SParty(alice)),
+          Set(alice),
+        )
+        inside(res) { case Success(Left(SErrorDamlException(IE.Limit(IE.Limit.ValueNesting(_))))) =>
+          msgs shouldBe Seq("starts test", "maintainers")
+        }
+      }
+    }
+
+    "lookup_by_key" - {
+
+      "a non-cached global contract" - {
+
+        // TEST_EVIDENCE: Semantics: Evaluation order of successful lookup_by_key of a non-cached global contract
```

Comment:
is it worth testing both a lookup by key that returns `Some` and one that returns `None`?